### ISSUE TITLE
Retval[0] has already been set 

### ISSUE THREAD

**Issue by DecentGradient** [tensorflow/tensorflow#24603]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Modified Code
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Fedora 28
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):binary pip
- TensorFlow version (use command below):1.12
- Python version:3
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

You can collect some of this information using our environment capture [script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
 I have a working SavedModel (ie. a saved model that works when restored in python) **that fails when run on tensorflow serving.**
The error message on the server is:
OP_REQUIRES failed at function_ops.cc:68 : Internal: Retval[0] has already been set.
Error: W **external/org_tensorflow/tensorflow/core/framework/op_kernel.cc:1401]** OP_REQUIRES failed at function_ops.cc:68 : Internal: Retval[0] has already been set.

 
### Exact Steps to Reproduce
(https://drive.google.com/file/d/1at1CQ9iHgcPHCn-MkvSGcgtbVM2lrKJn/view) link to saved model. it can be restored and run in python successfully but will error if run on a model server. (Takes an image as input  sess.run(fetches=["loop/Exit_1:0"],feed_dict={"image_bytes:0": image})

### Source code / logs
Relevant source code(I hope):
(contains a while loop with a concat in the body)
`
val, idx =tf.nn.top_k(softmax ,name="topk")
    sentence = tf.Variable([vocab.start_id],False,name="sentence",)
    sentence = tf.concat([sentence, idx[0]], 0)#

    def cond(sentence,state):
        return tf.math.not_equal( sentence[-1],tf.constant(vocab.end_id))

    def body(sentence,state):
        input_seqs = tf.expand_dims([sentence[-1]], 1)

        seq_embeddings = tf.nn.embedding_lookup(self.embedding_map, input_seqs)
        embed = seq_embeddings

        # In inference mode, use concatenated states for convenient feeding and
        # fetching.
        state_feed = tf.concat(axis=1, values=state, name="state")

        # Placeholder for feeding a batch of concatenated states.
        # state_feed = tf.placeholder(dtype=tf.float32,
        #                             shape=[None, sum(lstm_cell.state_size)],
        #                             name="state_feed")
        state_tuple = tf.split(value=state_feed, num_or_size_splits=2, axis=1)

        # Run a single LSTM step.
        lstm_outputs, new_state_tuple = lstm_cell(
            inputs=tf.squeeze(embed, axis=[1]),
            state=state_tuple)

        # Concatentate the resulting state.
        state = tf.concat(axis=1, values=new_state_tuple, name="state")

        # Stack batches vertically.

        lstm_outputs = tf.reshape(lstm_outputs, [-1, lstm_cell.output_size])

        with tf.variable_scope("logits") as logits_scope:
            logits = tf.contrib.layers.fully_connected(
            inputs=lstm_outputs,
            num_outputs=self.config.vocab_size,
            activation_fn=None,
            weights_initializer=self.initializer,
            scope=logits_scope, reuse = True
            )

        softmax = tf.nn.softmax(logits, name="softmax")
        self.softmax = softmax
        val, idx = tf.nn.top_k(softmax, name="topk")

        sentence = tf.concat([sentence,idx[0]],0)
        self.output = sentence
        return [sentence, state]
    out = tf.while_loop(cond, body, [sentence, state],parallel_iterations=1,maximum_iterations=20,name="loop",shape_invariants=[tf.TensorShape([None]),tf.TensorShape([None,None])])

    return out
`
fails with error:
`
 W external/org_tensorflow/tensorflow/core/framework/op_kernel.cc:1401] OP_REQUIRES failed at function_ops.cc:68 : Internal: Retval[0] has already been set.

`

**Comment by ymodak** [comment]
Apologies for the delay in response. This issue is more suitable on TensorFlow Serving repo. Can you please post it on TensorFLow Serving repo from [here](https://github.com/tensorflow/serving/issues). Thanks!
